vue-app/src/components/CopyButton.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <tooltip
+    v-if="text"
+    :position="position || 'bottom'"
+    :content="isCopied ? 'Copied!' : `Copy${type && ` ${type}`}`"
+  >
+    <div
+      :class="`${myClass || 'default'} ${hasBorder && 'border'}`"
+      @click="copyToClipboard"
+    >
+      <img width="16px" src="@/assets/copy.svg" />
+    </div>
+  </tooltip>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import Component from 'vue-class-component'
+import { Prop } from 'vue-property-decorator'
+
+import Tooltip from '@/components/Tooltip.vue'
+
+@Component({ components: { Tooltip } })
+export default class CopyButton extends Vue {  
+  @Prop() text!: string // Required: Text to copy
```

Comment:
I would use `value` as the name of this prop for consistency with HTML elements.